### PR TITLE
Add Azure ML Callback for experiment tracking

### PR DESCRIPTION
**Pull request by FabianHertwig**
This adds a Callback to track experiments with Azure ML

I basically copied and adapted the Neptune Callback, therefore I ran into the same issue as described in #3187. This should be fixed once I know what the solution is.

### CODE REVIEW

File: fastai/callback/azureml.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from ..basics import *
+from ..learner import Callback
+
+# Cell
+from azureml.core.run import Run
+
+# Cell
+class AzureMLCallback(Callback):
+    "Log losses, metrics, model architecture summary to AzureML"
+
```
Comment (jph00) [inline fastai/callback/azureml.py:16]: Add this line here to fix your smooth loss problem:

```python
order = Recorder.order+1
```

**Review by jph00 — CHANGES_REQUESTED** [review]
Many thanks - just one suggestion to fix your bug

File: fastai/callback/azureml.py (hunk 1 above)
Comment (FabianHertwig) [reply fastai/callback/azureml.py:16]: Do you mean `self.order`? Just setting order did not work.

Comment (FabianHertwig) [reply fastai/callback/azureml.py:16]: Ah, I got that wrong. It is set just inside the class and not in a `__init__`. 

Will add the commit in the next minutes.

**Comment by jph00** [comment]
Many thanks.

**Pull request merged**